lib/widgets/message_list.dart

```diff
@@ -781,58 +783,86 @@ class MessageWithPossibleSender extends StatelessWidget {
+          ]));
+    } else {
+      senderWidget = const SizedBox.shrink();
+    }
+
     return GestureDetector(
       behavior: HitTestBehavior.translucent,
       onLongPress: () => showMessageActionSheet(context: context, message: message),
-      // TODO clean up this layout, by less precisely imitating web
-      child: Padding(
-        padding: const EdgeInsets.only(top: 2, bottom: 3, left: 8, right: 8),
-        child: Row(crossAxisAlignment: CrossAxisAlignment.start, children: [
-          item.showSender
-            ? Padding(
-                padding: const EdgeInsets.fromLTRB(3, 6, 11, 0),
-                child: GestureDetector(
-                  onTap: () => Navigator.push(context,
-                    ProfilePage.buildRoute(context: context,
-                      userId: message.senderId)),
-                  child: Avatar(size: 35, borderRadius: 4,
-                    userId: message.senderId)))
-            : const SizedBox(width: 3 + 35 + 11),
-          Expanded(
-            child: Column(
-              crossAxisAlignment: CrossAxisAlignment.stretch,
-              children: [
-                if (item.showSender) ...[
-                  const SizedBox(height: 3),
-                  GestureDetector(
-                    onTap: () => Navigator.push(context,
-                      ProfilePage.buildRoute(context: context,
-                        userId: message.senderId)),
-                    child: Text(message.senderFullName, // TODO get from user data
-                      style: const TextStyle(fontWeight: FontWeight.bold))),
-                  const SizedBox(height: 4),
-                ],
-                MessageContent(message: message, content: item.content),
-                if ((message.reactions?.total ?? 0) > 0)
-                  ReactionChipsList(messageId: message.id, reactions: message.reactions!)
-              ])),
-          Container(
-            width: 80,
-            padding: const EdgeInsets.only(top: 4, right: 16 - 8),
-            alignment: Alignment.topRight,
-            child: Text(time, style: _kMessageTimestampStyle)),
-        ])));
+      child: Column(
+        children: [
+          senderWidget,
+          Padding(
+            padding: const EdgeInsets.symmetric(vertical: 4),
+            child: Row(crossAxisAlignment: CrossAxisAlignment.start, children: [
+              const SizedBox(width: 16),
+              Expanded(
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.stretch,
+                  children: [
+                    MessageContent(message: message, content: item.content),
+                    if ((message.reactions?.total ?? 0) > 0)
+                      ReactionChipsList(messageId: message.id, reactions: message.reactions!)
+                  ])),
+              SizedBox(width: 16,
+                child: message.flags.contains(MessageFlag.starred)
+                  ? Icon(ZulipIcons.star,
```

Comment:
This star appears at the top right of the message. In Vlad's design it looks like it's at the bottom right:
![image](https://github.com/zulip/zulip-flutter/assets/28173/1512fd21-1584-4a75-afd7-bc6e046655dd)

I guess I'm not 100% sure that's meant as one message — maybe the idea is that that's two messages, and the star is at the top of the second one. @terpimost, can you confirm which you intended?

Reply:
Coming off of 
<img width="980" alt="193037429-aeb6af5d-7d93-43db-bcc9-809ddbab5fd7" src="https://github.com/zulip/zulip-flutter/assets/98299/4f1c025d-fe35-4cb6-902e-ca18fbafd510">
 from https://github.com/zulip/zulip-mobile/issues/5511 it shows the star is aligned at the top.

Reply:
Great, sounds good.

Would you post a screenshot from this PR where the text is long enough that it gets near the right edge? That would help in comparing the vertical alignment of the content vs. the star.